ocaml/libs/xapi-stdext/lib/xapi-stdext-threads/delay_stubs.c

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (C) 2024 Cloud Software Group
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ */
+
+#include <caml/memory.h>
+#include <caml/threads.h>
+#include <caml/custom.h>
+#include <caml/unixsupport.h>
+
+#include <stdbool.h>
+#include <errno.h>
+#include <pthread.h>
+
+typedef struct delay {
+	pthread_mutex_t mtx;
+	pthread_cond_t cond;
+	bool signaled;
+} delay;
+
+// Initialize delay
+// Returns error number or 0 if success
+static int delay_init(delay *d)
+{
+	int err;
+	pthread_condattr_t cond_attr;
+
+	d->signaled = false;
+
+	err = pthread_condattr_init(&cond_attr);
+	if (err)
+		goto err0;
+	err = pthread_condattr_setclock(&cond_attr, CLOCK_MONOTONIC);
+	if (!err)
+		err = pthread_cond_init(&d->cond, &cond_attr);
+	if (err)
```

Comment:
`pthread_condattr_setclock` doesn't need a special cleanup? I'm wondering why this is not an else if